subsys/bluetooth/mesh/adv.h

```diff
@@ -75,18 +77,22 @@ int bt_mesh_adv_start(const struct bt_le_adv_param *param, int32_t duration,
 		      const struct bt_data *sd, size_t sd_len);
 
 static inline void bt_mesh_adv_send_start(uint16_t duration, int err,
-					  const struct bt_mesh_send_cb *cb,
-					  void *cb_data)
+					  struct bt_mesh_adv *adv)
 {
-	if (cb && cb->start) {
-		cb->start(duration, err, cb_data);
+	if (adv->cb && adv->cb->start && !adv->started) {
+		adv->started = 1;
+		adv->cb->start(duration, err, adv->cb_data);
+
+		if (err) {
+			adv->cb = NULL;
+		}
```

Comment:
Why not add `&& adv->started` to the `if` in `bt_mesh_adv_send_end` instead? It seems like it would be slightly easier to follow. This, and the `if (started)` in the destructor could be removed.

Reply:
There's no need for this if anymore

Reply:
We should deal with the error situation. What do you mean?

Reply:
Riiight, I see. I was thinking that the `started` flag should only be set if the advertising could actually be started, ie, it did not return an error. This works as well though, so I'll approve, and leave it to you :)